src/modules/popup/newPopup.js

```diff
@@ -0,0 +1,123 @@
+import allComments from '../getAllComments.js';
+import newComment from '../newComment.js';
+import { closePopup } from './popup.js';
+
+const updateCommentsSection = (comments, commentsDiv) => {
+  commentsDiv.innerHTML = ''; // Clear the container before adding updated comments
+  if (comments.length > 0) {
+    comments.forEach((comment) => {
+      const commentItem = document.createElement('div');
+      commentItem.classList.add('comment-item');
+      commentItem.innerHTML = `
+        <div class='commentaire'>
+          <p class="username">
+            <span class="date">${comment.creation_date}</span>
+            ${comment.username}: <span class="comment-text">${comment.comment}</span>
+          </p>
+        </div>
+      `;
+      commentsDiv.appendChild(commentItem);
+    });
+  } else {
+    const noCommentsMessage = document.createElement('p');
+    noCommentsMessage.textContent = 'No comments yet.';
+    commentsDiv.appendChild(noCommentsMessage);
+  }
+};
+
+const showPopup = async (mealId) => {
+  const blurr = document.querySelector('#overlay');
+  blurr.classList.add('overlay');
+  const popupUrl = `https://www.themealdb.com/api/json/v1/1/lookup.php?i=${mealId}`;
+  try {
+    const response = await fetch(popupUrl);
+    const data = await response.json();
+    const comments = await allComments(mealId);
+    // Check if data.meals is not null and contains at least one meal
+    if (data.meals && data.meals.length > 0) {
+      const meal = data.meals[0];
+
+      // Create the popup content
+      const commentPopup = document.createElement('div');
+      commentPopup.innerHTML = `
+        <div class='popup'>
+          <span class="close" id="close">&times;</span>
+          <div class='image'>
+            <img id="popup-image" src="${meal.strMealThumb}" alt="${meal.strMeal}">
+          </div>
+          <h3 id="popup-title">${meal.strMeal}</h3>
+          <ul id="foodDetail" class="food-detail">
+            <li><strong>Area:</strong> ${meal.strArea} </li>
+            <li><strong>Category:</strong> ${meal.strCategory} Pizza</li>
+          </ul>
+          <hr>
+          <h3 id ="cmt"> Comments <span id="comments-count">(${comments.length})</span> </h3>
```

Comment:
As mentioned in the [requirements](https://github.com/microverseinc/curriculum-javascript/blob/main/group-capstone/js_capstone.md#:~:text=Even%20if%20the%20API%20gives%20you%20these%20numbers%2C%20you%20will%20create%20a%20specific%20function%20to%20calculate%20these%20numbers%20in%20each%20page.%20These%20count%20functions%20need%20to%20be%20covered%20with%20unit%20tests%20using%20Jest.), you should use a function to calculate the number of comments displayed and not rely on what the API returns.